runtime/vm/stringhelpers.cpp

```diff
@@ -415,38 +415,49 @@ getStringUTF8Length(J9VMThread *vmThread, j9object_t string)
 UDATA 
 verifyQualifiedName(J9VMThread *vmThread, j9object_t string)
 {
+	J9JavaVM* vm = vmThread->javaVM;
 	UDATA unicodeLength = J9VMJAVALANGSTRING_LENGTH(vmThread, string);
 	j9object_t unicodeBytes = J9VMJAVALANGSTRING_VALUE(vmThread, string);
-	UDATA i = 0;
+	J9CfrConstantPoolInfo classNameInfo = {0, 0, 0, 0, 0, NULL, 0};
 	UDATA rc = CLASSNAME_VALID_NON_ARRARY;
+	UDATA i = 0;
+	IDATA arity = 0;
 
-	if (unicodeLength == 0) {
+	PORT_ACCESS_FROM_JAVAVM(vm);
+
+	if (0 == unicodeLength) {
 		return CLASSNAME_INVALID;
 	}
 
+	classNameInfo.slot1 = (U_32)unicodeLength;
+	classNameInfo.bytes = (U_8*)j9mem_allocate_memory(unicodeLength, OMRMEM_CATEGORY_VM);
+
 	if (IS_STRING_COMPRESSED(vmThread, string)) {
 		if ('[' == J9JAVAARRAYOFBYTE_LOAD(vmThread, unicodeBytes, 0)) {
 			rc = CLASSNAME_VALID_ARRARY;
 		}
 		for (i = 0; i < unicodeLength; i++) {
-			U_8 c = J9JAVAARRAYOFBYTE_LOAD(vmThread, unicodeBytes, i);
-			if (c == '/') {
-				rc = CLASSNAME_INVALID;
-				break;
-			}
+			classNameInfo.bytes[i] = J9JAVAARRAYOFBYTE_LOAD(vmThread, unicodeBytes, i);
 		}
 	} else {
 		if ('[' == J9JAVAARRAYOFCHAR_LOAD(vmThread, unicodeBytes, 0)) {
 			rc = CLASSNAME_VALID_ARRARY;
 		}
 		for (i = 0; i < unicodeLength; i++) {
-			U_16 c = J9JAVAARRAYOFCHAR_LOAD(vmThread, unicodeBytes, i);
-			if (c == '/') {
-				rc = CLASSNAME_INVALID;
-				break;
-			}
+			classNameInfo.bytes[i] = (U_8)J9JAVAARRAYOFCHAR_LOAD(vmThread, unicodeBytes, i);
```

Comment:
This loads a `U_16` and then throws away 1/2 of it by casting to `U_8`

Reply:
The code involved there has been replaced with copyStringToUTF8WithMemAlloc.